timesketch/lib/datastores/opensearch.py

```diff
@@ -1271,3 +1296,505 @@ def flush_queued_events(self, retry_count=0):
+                os_logger.error(
+                    "[Slice %s/%s] Failed to put sentinel: Queue full.",
+                    log_slice_id,
+                    num_slices,
+                )
+            except Exception as e_s:  # pylint: disable=broad-exception-caught
+                os_logger.error(
+                    "[Slice %s/%s] Failed to put sentinel: %s",
+                    log_slice_id,
+                    num_slices,
+                    str(e_s),
+                )
+
+    def export_events_with_slicing(
+        self,
+        indices_for_pit: List[str],
+        base_query_body: Dict[str, Any],
+        sort_criteria: Optional[List[Dict[str, Any]]] = None,
+        page_size: Optional[int] = None,
+        pit_keep_alive: Optional[str] = None,
+        num_slices: Optional[int] = None,
+        request_timeout_per_slice: Optional[int] = None,
+    ) -> Generator[Dict[str, Any], None, None]:
+        """
+        Exports events from specified indices using OpenSearch Point-In-Time (PIT),
+        search_after, and slicing. PIT is used internally to ensure a consistent
+        view of the data across multiple paginated and sliced requests.
+
+        This method is designed for efficient bulk export of large datasets.
+        It yields individual event dictionaries, including '_id' and '_index'.
+
+        The following parameters can be overridden by the caller. If not provided,
+        they will default to values from the Timesketch configuration or
+        sensible internal defaults:
+            - `page_size`: Defaults to `OPENSEARCH_SLICED_EXPORT_DEFAULT_PAGE_SIZE`
+            - `pit_keep_alive`: Defaults to `OPENSEARCH_SLICED_EXPORT_PIT_KEEP_ALIVE`
+            - `num_slices`: Defaults to `OPENSEARCH_SLICED_EXPORT_NUM_SLICES` or 4
+            - `request_timeout_per_slice`: Defaults to
+              `OPENSEARCH_SLICED_EXPORT_REQUEST_TIMEOUT` or a calculated value
+              based on the general OpenSearch timeout +20s.
+
+        The internal queue size is determined by `page_size`, `num_slices`, and
+        the `OPENSEARCH_SLICED_EXPORT_QUEUE_BUFFER_FACTOR` configuration.
+
+        Args:
+            indices_for_pit: List of index names to query.
+            base_query_body: The base OpenSearch query body (must not include
+                             size, sort, pit, or slice clauses). Can include
+                             "_source" to specify fields to return.
+            sort_criteria: Optional. The sort criteria for search_after.
+                           If None, defaults to `[{"_id": "asc"}]`.
+                           Must include a unique tie-breaker field (e.g. _id)
+                           for stable results across pages.
+            page_size: Optional. Number of documents per request per slice.
+                       Overrides the global default (10_000) if provided.
+            pit_keep_alive: Optional. Keep-alive duration for the Point-In-Time context
```

Comment:
since this is a string, maybe explain a bit more in what format it should be provided?